Add example filetype schema created with LinkML

ml-evs opened the pull request:
Here's an example of a filetype schema made using [LinkML](https://linkml.io/linkml/), a data modelling language that can be exported to many schema formats (JSONSchema), runtime models (pydantic) and linked data formats (RDF etc). 

This could be used to populate the registry, and also aid development of the input schema for extractor code.

Usage:

```shell
pip install -r requirements.txt # (just LinkML for now)
# validate example for MPR (currently gives an error I don't yet understand)
linkml-validate -s filetype.yml -C FileType biologic_mpr.yml
# generate a JSONSchema to be used elsewhere
gen-json-schema filetype.yml
# auto-generate pydantic code for use within Python applications
gen-pydantic filetype.yml
# generate Java class models
gen-java filetype.yml
```

ml-evs committed:
Add example filetype schema created with LinkML

PeterKraus commented on filetype.yml:
```diff
@@ -0,0 +1,36 @@
+id: marda_filetype_schema
+default_range: string
+prefixes:
+  schema: http://schema.org
```
`schema -> schema.org`

ml-evs commented on filetype.yml:
```diff
@@ -0,0 +1,36 @@
+        slot_uri: schema:name
+      description:
+        slot_uri: schema:description
+      vendor:
```
```suggestion
      primary_vendor:
```
?

```suggestion
   associated_vendors: 
      multivalued: true
````

PeterKraus replied:
I think `associated_vendors` is perhaps more descriptive.

ml-evs commented on filetype.yml:
```diff
@@ -0,0 +1,36 @@
+      associated_software:
+        description: >-
+          Any software (proprietary or otherwise) that produces such files.
+      subformats:
+        multivalued: true
+        range: FileType
+        inlined_as_list: true
```
```suggestion
      subformats:
        multivalued: true
        range: FileType
        inlined_as_list: true
        
   associated_standards:
       multivalued: true
```

PeterKraus replied:
Is the plan to keep both `subformats` and `associated_standards`? I might have misunderstood.

ml-evs replied:
No sorry, bad suggestion comment that I posted rather than losing it. Just pushing stuff now.

ml-evs committed:
Update schema and example

ml-evs committed:
Trying to make schema understandable to LinkML

ml-evs commented:
Still getting some LinkML errors from this (you have to do some imports within the schema to load things properly, haven't quite got the right incantation yet) but this is probably ready to look at again @PeterKraus before merging

ml-evs commented:
I can perhaps add a note to the README too outlining the purpose of this demo

PeterKraus commented on biologic_mpr.yml:
```diff
@@ -0,0 +1,17 @@
+id: >-
+  biologic-mpr
```
Minor thing, but I'd prefer `biologic.mpr` as it makes the suffix a little bit more obvious.

ml-evs replied:
My only concern here is that these IDs will probably be used for canonical URLs at some point, e.g., `https://registry.marda-alliance.org/filetypes/biologic-mpr` so I guess "." isn't URL safe

PeterKraus replied:
Noted.

PeterKraus commented on biologic_mpr.yml:
```diff
@@ -0,0 +1,17 @@
+name: >-
+  Biologic MPR file
+description: >-
+  A binary data file created by Biologic cyclers, containing time- and frequency-domain data pertaining to many electrochemical techniques, such as voltammetry, potentiometry, coulometry and impedance spectroscopy.
```
Question about "code" style: Should we be running those files through `black` or some other formatter on commit? Or are lines of arbitrary length, such as those descriptions, OK?

ml-evs replied:
I was thinking of adding `yamlfmt` via pre-commit CI at a later date, but can just do it now

PeterKraus commented on biologic_mpr.yml:
```diff
@@ -0,0 +1,17 @@
+  A binary data file created by Biologic cyclers, containing time- and frequency-domain data pertaining to many electrochemical techniques, such as voltammetry, potentiometry, coulometry and impedance spectroscopy.
+associated_vendors: >-
+  - Biologic
+subject: ->
```
I'm not clear on the syntax here, but what's the difference between `>-` and `->`?

ml-evs replied:
One is a typo, the other isn't :)

PeterKraus commented on biologic_mpr.yml:
```diff
@@ -0,0 +1,17 @@
+id: >-
+  biologic-mpr
+name: >-
+  Biologic MPR file
```
I think it's `BioLogic`, by the way.

PeterKraus reviewed (approved):
Looks good. Once you figure out how to "compile" it using linkml, I think it's good to go.

ml-evs committed:
Fix typos and add yaml checker via pre-commit

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ml-evs committed:
Bump pre-commit hook version

ml-evs committed:
Fix schemas for LinkML generators: needed to remove `inlined_as_list`

ml-evs committed:
Update README and fix final typos

ml-evs commented:
I've addressed all your issues and found the source of the bug (need to understand more what `inline_as_list` does, as it only works without it...).

I'll merge this now and we can use this is as a starting point for further discussions.

ml-evs committed:
README formatting
